### PR TITLE
03-JavaServiceSkeleton

### PR DESCRIPTION
**Pull request by d-serj**
Impelented simple service to store string. Implemented 2 applications to receive and transmit string from/to the service

### CODE REVIEW

File: 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    strOutput.setText(output);
+                } catch (StringIndexOutOfBoundsException | RemoteException | NullPointerException e) {
+                    Log.e("StringsReceiver", "string receive exception!\n");
+                }
+            }
+        });
+
+        Intent mIntent = new Intent();
+        mIntent.setClassName("com.example.stringservice", "com.example.stringservice.StringService");
+        bindService(mIntent, mConnection, BIND_AUTO_CREATE);
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java]: Where is unbindService() ?

File: 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        btnReceive.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String output;
+
+                try {
+                    output = mService.stringReceive();
+                    strOutput.setText(output);
+                } catch (StringIndexOutOfBoundsException | RemoteException | NullPointerException e) {
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java]: What is StringIndexOutOfBoundsException for here?

File: 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        final Button transmitStr = (Button)findViewById(R.id.transmitStr);
+
+        transmitStr.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String input = strField.getText().toString();
+
+                try {
+                    mService.stringTransmit(input);
+                } catch (StringIndexOutOfBoundsException | RemoteException | NullPointerException e) {
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java]: What is StringIndexOutOfBoundsException for here?

File: 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    Log.e("StringsTransmitter", "string transmit exception!\n");
+                }
+            }
+        });
+
+        Intent mIntent = new Intent();
+        mIntent.setClassName("com.example.stringservice", "com.example.stringservice.StringService");
+        bindService(mIntent, mConnection, BIND_AUTO_CREATE);
```
Comment (BulldozerBSG) [inline 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java]: Where is unbindService() ?

**Review by BulldozerBSG — CHANGES_REQUESTED** [review]

File: 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java (hunk 1 above)
Comment (d-serj) [reply 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java]: unbindService() was added

File: 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java (hunk 4 above)
Comment (d-serj) [reply 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java]: unbindService() was added

File: 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java (hunk 3 above)
Comment (d-serj) [reply 03-JavaServiceSkeleton/StringsTransmitter/app/src/main/java/com/example/stringstransmitter/MainActivity.java]: Removed

File: 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java (hunk 2 above)
Comment (d-serj) [reply 03-JavaServiceSkeleton/StringsReceiver/app/src/main/java/com/example/stringsreceiver/MainActivity.java]: Removed

**Review by BulldozerBSG — APPROVED** [review]

**Pull request merged**